File: 2379. Minimum Recolors to Get K Consecutive Black Blocks/MinimumRecolorsToGetKConsecutiveBlackBlocks.java
public class MinimumRecolorsToGetKConsecutiveBlackBlocks {
    public int minimumRecolors(String blocks, int k) {
        int count=0,sol,n = blocks.length();
        for(int i = 0;i<k; i++){
            if(blocks.charAt(i)=='W'){
                count++;
            }
        }
        sol = count;
        for(int i = k;i<n;i++){
            if(blocks.charAt(i)=='W'){
                count++;
            }
            if(blocks.charAt(i-k)=='W'){
                count--;
            }
            sol = Math.min(count,sol);
        }
        return sol;
    }
    public static void main(String[] args) {
        MinimumRecolorsToGetKConsecutiveBlackBlocks mBlocks = new MinimumRecolorsToGetKConsecutiveBlackBlocks();
        System.out.println(mBlocks.minimumRecolors("WWBWBWBWBW", 2));
    }
}

File: 2379. Minimum Recolors to Get K Consecutive Black Blocks/minimum recolors to get k consecutive black blocks.py
blocks = "BBWBWBWBW"
k = 2
index = 0
n = len(blocks)
count = 0
for i in range(k):
    if blocks[i]=="W":
        count+=1
sol = count
while (k<n):
    if blocks[k]=="W" and blocks[index] == "B":
        count+=1
    elif blocks[k]=="B" and blocks[index] == "W":
        count-=1
    k+=1
    index+=1
    sol = min(count,sol)
print(sol)